### PR TITLE
Clean up and update tests

### PR DESCRIPTION
**Pull request by MarkusH**
## Summary of the changes / Why this is an improvement

TL;DR of changes:

- Dropped all usages of `unittest.TestCase` in favor of py.test functions
- Moved a bunch of tests around; essentially, tests for commands will now go to `tests/test_<toplevel-command>.py`. Naming of the test functions should follow the pattern of `test_<1st-level-command>_<2nd-level-command>_...()`, e.g. `test_me`, `test_users_list`, `test_organizations_users_add`. This eases the process of running only some tests using py.test's `-k` flag:
  ```shell
  $ tox -e py37 -- -k test_organizations_users_
  ```
- Fixed side-effects / broken server, REST Client, and HttpSession tests

## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by MarkusH** [comment]
Closed as part of #234
